### PR TITLE
[Merged by Bors] - chore(Data/Set/Image): @[simp]

### PR DESCRIPTION
**Pull request by kim-em**
---

- [x] depends on: #6297

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Data/Set/Image.lean [hunk 1]
```diff
@@ -488,16 +488,19 @@ theorem subset_preimage_image (f : α → β) (s : Set α) : s ⊆ f ⁻¹' (f '
...
 
+@[simp]
 theorem image_preimage_eq {f : α → β} (s : Set β) (h : Surjective f) : f '' (f ⁻¹' s) = s :=
   Subset.antisymm (image_preimage_subset f s) fun x hx =>
     let ⟨y, e⟩ := h x
     ⟨y, (e.symm ▸ hx : f y ∈ s), e⟩
 #align set.image_preimage_eq Set.image_preimage_eq
 
+@[simp]
 theorem preimage_eq_preimage {f : β → α} (hf : Surjective f) : f ⁻¹' s = f ⁻¹' t ↔ s = t :=
```
Comment (eric-wieser) [inline Mathlib/Data/Set/Image.lean:504]: In Lean 3 I vaguelly remember removing a bunch of conditional `simp` lemmas that used `injective`/`surjective` for performance reasons. I don't remember the details though, and it's very possible that they're much less of a problem in Lean 4. Perhaps worth benchmarking?

**Comment by semorrison** [comment]
!bench

**Comment by semorrison** [comment]
!bench

**Comment by semorrison** [comment]
Okay, I've now rebased onto something that has already been benchmarked, and will try again!

**Comment by semorrison** [comment]
!bench

File: Mathlib/Data/Set/Image.lean (hunk 1 above)
Comment (j-loreaux) [reply Mathlib/Data/Set/Image.lean:504]: The benchmarked change for time spent in `simp` was `+0.014 %`, so I don't think we need to worry about it.

**Comment by j-loreaux** [comment]
Thanks!

bors r+

**Pull request closed without merging**